File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        // create map of sorted_str : array of anagrams
        // store temp, sort, add to ans

        std::unordered_map<std::string,std::vector<std::string>> groups;


        std::string temp;
        for(auto str: strs){
            temp = str;
            std::sort(str.begin(),str.end());
            groups[str].push_back(temp);
        }


        std::vector<std::vector<std::string>> ans; 
        for(auto pair:groups){
            ans.push_back(pair.second);
        }

        return ans;

    }
};

File: 0049-group-anagrams/0049-group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    
    // iterate through strs
    // sort current anagram string, check key value pairs for matching
    // if found, add to that array
    // if not found, add it as key value with empty array

    // Map => {string: string[]  }

    const myMap = new Map()




    for(let i = 0;i < strs.length;i++){

        let sortedStr = strs[i].split('').sort().join('')

        if(myMap.has(sortedStr)){

            myMap.get(sortedStr).push(strs[i])

        }else{
            myMap.set(sortedStr,[])
            myMap.get(sortedStr).push(strs[i])
        }
    } 

    let anagrams = []

    myMap.forEach((value, key)=>{

        anagrams.push(value)

    })


    return anagrams

};